src/D2L.Security.OAuth2/Validation/Request/RequestAuthenticator.cs

```diff
@@ -19,10 +19,10 @@ HttpRequestMessage request
 		) {
 			string bearerToken = request.GetBearerTokenValue();
 
-			return AuthenticateHelper( bearerToken );
+			return AuthenticateAsync( bearerToken );
 		}
 
-		private async Task<ID2LPrincipal> AuthenticateHelper(
+		public async Task<ID2LPrincipal> AuthenticateAsync(

```

Comment:
.NET 5 has `HttpRequest` from [a NuGet package](https://www.nuget.org/packages/Microsoft.AspNetCore.Http/).

I could add an overload, but then we would need to dep on this package for the .NET 5 build. This sucks because the dependency is maybe not necessary in some cases (imagine a Lambda function taking tokens in its request body for whatever reason). It mostly sucks because its another dependency to rev in this DLL which makes upgrades more annoying.

Previously we had a separate ASP.NET integration DLL. I like that pattern but stuff is a bit whack right now in the transition.

Since `IAccessTokenValidator` is public I don't see any reason to not make this one public.